### ISSUE TITLE
CUDA error while runing unsupervised_TU on

### ISSUE THREAD

**Issue by c747903646** [Shen-Lab/GraphCL#42]

**Comment by yyou1996** [comment]
Hi @c747903646,

We didn't conduct unsupervised_TU on GITHUB dataset (but do in semisupervised_TU). It looks like the edge index get mismatches when `x = F.relu(self.convs[i](x, edge_index))` is executed. Would you confirm that via (i) trying other datasets e.g. NCI1 with the same script, and (ii) running on GITHUB dataset semisupervised setting?

**Comment by c747903646** [comment]
> Hi @c747903646,
> 
> We didn't conduct unsupervised_TU on GITHUB dataset (but do in semisupervised_TU). It looks like the edge index get mismatches when `x = F.relu(self.convs[i](x, edge_index))` is executed. Would you confirm that via (i) trying other datasets e.g. NCI1 with the same script, and (ii) running on GITHUB dataset semisupervised setting?

Yes, I have run the same script on other TU datasets, such as COLLAB, DD, NCI1, RDT-5K and RDT-B, and all work except run on GITHUB. Meanwhile, semisupervised codes work all TU datasets, including GITHUB.

In debugging, I found that before `x = F.relu(self.convs[i](x, edge_index))` is excuted, the dimension of `x` is matched with `self.convs[i]`. So I could not understand the error info, `srcIndex < srcSelectDimSize`.

**Comment by yyou1996** [comment]
Mismatch could happen if `edge_index.max()+1 < x.shape`, which is my suspect.
